### DISCUSSION TITLE
能否解释一下为何使用 useRef 在 useObservableCallback 中会撕裂？

### DISCUSSION THREAD

**Discussion by 07akioni** [crimx/observable-hooks#80] [Q&A]
不是很理解。

是因为某次 reconcile 过程中 ref 的值已经被修改了，但是之后却被打断了，然后 effect 还在用老的状态？

**Comment by crimx** [comment]
你好，在 React 并行模式下，render 会出现暂停的情况，这时如果 ref 对象被修改，就会出现不一致的问题。具体到 useSubscription 中，如果暂停过程中 Observable 被换掉，则 Observer 在切换之前有可能拿到旧 Observable 输出的值。
